### PR TITLE
Add blog support

### LINKED ISSUE #4: Create blog layout
**Is this a BUG REPORT or FEATURE REQUEST?**:
feature

**What happened**:
No blog page/post layout is available.

**What you expected to happen**:
Support for a blog/post layout.

**How to reproduce it (as minimally and precisely as possible)**:
Try to run a blog with syna.

**Environment**:
- Syna Theme version:  v0.7.0
- Hugo version:  v0.30.2

### PR DESCRIPTION
**Pull request by mpourismaiel**
**What this PR does / why we need it**:
Add simple blog support. This PR adds support for index files to be fragments themselves to make smaller pages easier to create. Also adds list fragment that is able to list pages from different sections and is used in the exampleSite to list latest articles.

**Which issue this PR fixes**:
fixes #4 

**Release note**:
```release-note
- Add new list fragment that lists pages from a single section
- Index files now can be fragments
```

### CODE REVIEW

File: layouts/partials/fragments/list.html [hunk 1]
```diff
@@ -0,0 +1,108 @@
+{{/*
+  This variable stores all the page bundles in the blog section. There are three
+  where functions achieving this. First where gets all the pages in blog
+  section, the second one removes current page from the mix and the last one
+  removes the section (.Site.Pages returns the current section as a page). */}}
+{{- $articles := where (where .Site.Pages "Section" (.Params.section | default .page.Section)) "Params.fragment" "in" (slice "content-single" "content-split") -}}
```
Comment (stp-ip) [inline layouts/partials/fragments/list.html]: That's is quite a great query. Wouldn't have thought of that.
It implicitly defaults to .page.Section. Does that always work? Or might there be some strange edge cases?

File: exampleSite/content/blog/_index/index.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
++++
+date = "2018-07-06"
+url = "/blog"
```
Comment (stp-ip) [inline exampleSite/content/blog/_index/index.md]: Not sure I like the `url = "/blog"` workaround. From an implementation perspective I see why, but from a user perspective it feels more complex than it should.
Another option would be to go with this options for now and in a separate release/step work on the defaulting.

Basically a section index, would automatically be a list fragment. Leaf bundle content would automatically be a content fragment.

File: layouts/partials/fragments/list.html [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    <div class="container py-5">
+      <l class="row">
+        <div class="col-12
+          {{- if or (eq $bg "secondary") (eq $bg "primary") -}}
+            {{- printf " text-%s" "dark" -}}
+          {{- else -}}
+            {{- printf " text-muted text-%s" "secondary" -}}
+          {{- end -}}
+        ">
+          {{- range first (.Params.article_count | default 10) $articles }}
```
Comment (stp-ip) [inline layouts/partials/fragments/list.html]: Could be shortened to just ".Params.count" I assume.

File: layouts/partials/fragments/list.html (hunk 1 above)
Comment (mpourismaiel) [reply layouts/partials/fragments/list.html]: Nothing that I'm aware of. The only thing is if the list fragment is used in the root of content directory or any first level directories there wouldn't be any sections. It's just the Hugo's definition of section.

File: exampleSite/content/blog/_index/index.md (hunk 2 above)
Comment (mpourismaiel) [reply exampleSite/content/blog/_index/index.md]: Changed in 9ac6dd681a5d1e8c85a0610be1fe0fe2465dad88 to `/blog/latest`

File: layouts/partials/fragments/list.html (hunk 3 above)
Comment (mpourismaiel) [reply layouts/partials/fragments/list.html]: Changed to `count` and also added default values in the example in b7caa1581b3a1ca548502229eef3e3efc828223e

File: exampleSite/content/blog/_index/index.md (hunk 2 above)
Comment (stp-ip) [reply exampleSite/content/blog/_index/index.md]: Sorry probably could have stated my point better.
I dislike the usage of `url =` as this is something the user have to change, while knowing how Hugo would interpret it. The location was actually correct in my view.
We have `/blog/article/` so the list should be located at `/blog/`. The users just shouldn't need to know that.

**Comment by stp-ip** [comment]
Adding it here as Github makes it hard to find comments, that were "fixed" via code:

> I dislike the usage of `url =` as this is something the user has to change, while knowing how Hugo would interpret it. The location `/blog/` was actually correct in my view.
> We have /blog/article/ so the list should be located at /blog/. The users just shouldn't need to know that.

**Comment by mpourismaiel** [comment]
By `/blog/` do you mean `exampleSite/content/blog/` or `example.com/blog/`? If you mean the first one (directory path) If I do so then all the other directories in that directory would be considered resources for the `example.com/blog/` unless we remove all directories in that path and use files. The only way hugo allows for index page in a section is by using list layout which we cannot support due to us wanting to support fragments and list layout has a weird way of resolving files.

**Comment by stp-ip** [comment]
example.com/blog/ was intended. For now it seems the workaround with `url=` is the best approach. I just wanted to mention, that it's not the best user UI. Sorry for the confusion.

**Comment by stp-ip** [comment]
LTGM and iterate in follow on PRs.

**Pull request merged**